src/main/java/com/minecolonies/coremod/colony/CitizenData.java

```diff
@@ -446,20 +447,38 @@ public void initializeFromEntity(@NotNull final EntityCitizen entity)
     private String generateName(@NotNull final Random rand)
     {
         String citizenName;
+        String firstName;
+        String middleInitial;
+        String lastName;
+
         if (female)
         {
-            citizenName = String.format("%s %s. %s", getRandomElement(rand, Configurations.names.femaleFirstNames), getRandomLetter(rand),
-              getRandomElement(rand, Configurations.names.lastNames));
+            firstName = getRandomElement(rand, Configurations.names.femaleFirstNames);
+        }
+        else
+        {
+            firstName = getRandomElement(rand, Configurations.names.maleFirstNames);
+        }
+
+        middleInitial = String.valueOf(getRandomLetter(rand));
+        lastName = getRandomElement(rand, Configurations.names.lastNames);
+
+        if(Configurations.names.useMiddleInitial) {
```

Comment:
You will need to update the ActionArgumentType citizen parser and tab completion code to ignore the middle initial.
I think the citizen action argument type also has it defined that a citizen name is 3 words.   I don't remember if that means "up to three words" or "three words exactly".  Easiest thing to do is to make the change, and add a test for a two-word citizen name.

Reply:
CommandEntryPointNew getTabCompletionsAndParsingHoldersForActionMenuTreeNode()  will try to parse up to actionArgumentType.allowedSpaceCount(), so that can be left as 3.

ActionArgumentType findCitizenForCitizenManager() is an exact match on the citizen name, so parsing doesn't need any changes.

ActionArgumentType.getCitizenTabCompletions() is all that will need fixing.  The fix for that looks straight-forward: if the config option is set, skip the middle+last-name-check and only do a last-name check against citizenNameParts[1]

And add tests for first+last citizen names when the config option is set to CommandEntryPointTest.

Reply:
Needs a more significant rewrite than that - the original ticket that spawned this was using "last names" that contained more than one word, such as "of Loxley". If you wanted to fully support all possibilities, I could set "Sally Jane" as a first name, and "Johnson the Third" as a last name.

That's assuming that anyone ever uses tab complete on a citizen name...how often are commands run against citizens rather than GUI interaction? What if you have a Fred Q Jones vs a Fred P Jones? (which is possible with the current code) Does anyone ever try to memorize citizen names when they die frequently? What if players rename citizens with name tags?

Reply:
It's a new feature, and I use tab completion to complete a citizen name.

Yes, Fred P. Jones vs Fred Q. Jones works under the current system.

What I said will be good enough for this ticket. 

For the future, though....

I didn't see the original ticket that spawned this code change.   However, tab completion in minecraft itself works on space-separated arguments.   So to support arbitrary-long sets of words as CItizen names, the rewrite will have to deal with each word of a citizen name being handled separately.

It's probably not as hard as it sounds.

We could likely get rid of allowedSpaceCount() since it's currently only used by citizen names, and switch over to parsing the maximum number of words possible, up to the maximum count from the current set of citizen names or the maximum count of the current set of remaining arguments to parse.

Reply:
We need some dynamic code to be able to deal with whatever length. At the moment due to the renaming feature it is possible to have citizen names of up to what? 10 separations.
R A Y C O M S would be valid for example.